### DISCUSSION TITLE
How to inference custom dataset without Folder()?

### DISCUSSION THREAD

**Discussion by cory0619** [open-edge-platform/anomalib#2357] [Q&A] [answered]
Hello all.

I've successfully trained my custom dataset!
However, I'm struggling to inference my test dataset.

I'd like to inference test dataset all but, when I'm using Folder() to inference test dataset as below, it leads to split my dataset.
`    datamodule = Folder(
        name='evaluation',
        root=str(evaluation_dir), 
        normal_dir=str(evaluation_dir), 
        task=TaskType.CLASSIFICATION,
        eval_batch_size=1,
    )`

I don't wanna split my test. then, I'd like to inference my dataset all.
Could you all help me?

Any method inference only 1 image?
I think, in order to inference only 1 image, Folder() should be deleted... right?
I don't understand why Folder() must have split ratio 0~1...

**Comment by AlexandreFyl** [comment]
Hello,

You can use TorchInferencer to achieve this, as follows :

`inferencer = TorchInferencer(path=model_path)`
`result= inferencer.predict(img_path)`


Your score will be available in the pred_score key !

**Reply by cory0619** [reply, marked as answer]
Thanks a lot. I got a solution as below.

```
    inferencer = TorchInferencer(path="D:\\patchcore\\anomalib_2\\models\\model.pt", device="cpu")

    image = Image.open("D:\\patchcore\\anomalib_2\\evaluation_1_image\\TEST_098.png").convert("RGB")
    image = image.resize((224, 224))
    transform = transforms.ToTensor()
    image_tensor = transform(image) 
    
    result = inferencer.predict(image=image_tensor)
```

**Reply by samet-akcay** [reply]
@cory619-seungwon, you may not even need the transforms, which are already within the model graph for the exported models. Just pass your input image, and the model should apply the transforms under the hood. You could refer to the following link for details.
https://anomalib.readthedocs.io/en/v1.1.1/markdown/guides/how_to/data/transforms.html

**Comment by cory0619** [comment]
Let me close my query. Many thanks to all.
